lib/queues.js

```diff
@@ -260,16 +270,64 @@ class Queue {
   runOnce() {
     assert(!this._processing, 'Cannot call once while continuously processing jobs');
     const hasHandlers = (this._handlers.size > 0);
+
     if (hasHandlers) {
 
       debug('Waiting for jobs on queue %s', this.name);
-      return this._getClientPromise.then(client => this._reserveAndProcess(client));
+
+      if (this._delayedJobs.size > 0) {
+
+        const jobIDs = this._getDelayedJobIDs();
+
+        return this._kickJobs(jobIDs)
```

Comment:
Wouldn't there still be an operation - kickJobs resolves the `_putClientPromise` before looping over the jobIDs collection...?